backend/e4h-services/ingestion-service/app/api/endpoints/file_ingestion.py

```
@@ -1699,6 +1701,32 @@ async def validate_facilities_excel_sheet(
 
         boundary_data_df = pd.read_excel(temp_input_file.name, sheet_name=boundary_sheet_name)
 
+        # ----------------- Validate Boundary Sheet Against Project ----------------- #
+        projects = project_client.search_project(request_info_obj, project_id)
+        if not projects or "Project" not in projects or len(projects["Project"]) == 0:
+            raise HTTPException(status_code=400, detail=f"No project found for id {project_id}")
```

Ruff (0.13.1)

Abstract `raise` to an inner function

(TRY301)